### ISSUE TITLE
Rework mnist training script

### ISSUE THREAD

**Issue by vladimirjovanovicTT** [tenstorrent/tt-forge-fe#806]
test_mnist_training() from forge/test/mlir/mnist/training/test_training.py should be reworked to showcase standard batched training.
It makes sense to have an implementation similar to the current one in order to test gradient accumulation - it just needs to be renamed and explained through comments.

**Comment by mnmnk43434** [comment]
Hello, we tried to solve the issue.

This is what we did:

Modified the test_mnist_training() function in forge/test/mlir/mnist/training/test_training.py to demonstrate standard batched training. Renamed the existing implementation to test_mnist_training_with_gradient_accumulation() to preserve the gradient accumulation test case.

You can review changes in this commit: https://github.com/mnmnk43434/tenstorrent-tt-forge-fe-806/commit/be5aab560724acfc774dc6d43ed8e74c8cdcbf6e.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.
